diff --git a/services/surfaceflinger/CompositionEngine/src/OutputLayer.cpp b/services/surfaceflinger/CompositionEngine/src/OutputLayer.cpp
index cc3a94c..2b5336c 100644
--- a/services/surfaceflinger/CompositionEngine/src/OutputLayer.cpp
+++ b/services/surfaceflinger/CompositionEngine/src/OutputLayer.cpp
@@ -970,9 +970,12 @@ void OutputLayer::detectDisallowedCompositionTypeChange(Composition from, Compos
     }
 
     if (!result) {
-        ALOGE("[%s] Invalid device requested composition type change: %s (%d) --> %s (%d)",
+        ALOGI("[%s] Overriding invalid composition change: %s (%d) --> %s (%d) for legacy compatibility",
               getLayerFE().getDebugName(), to_string(from).c_str(), static_cast<int>(from),
               to_string(to).c_str(), static_cast<int>(to));
+        result = true;
+        // Synchronize display state for Android 16
+        getLayerFE().setLastHwcState({.lastCompositionType = to});
     }
 }
 
